apiserver/pkg/storage/calico/resource.go

```diff
@@ -426,7 +426,7 @@ func (rs *resourceStore) GuaranteedUpdate(
 		revInt, _ := strconv.Atoi(accessor.GetResourceVersion())
 		updatedRes := updatedObj.(resourceObject)
 		if !shouldCreateOnUpdate() {
-			if updatedRes.GetObjectMeta().GetResourceVersion() == "" || revInt < int(curState.rev) {
+			if updatedRes.GetObjectMeta().GetResourceVersion() == "" || revInt < int64(curState.rev) {
```

Comment:
Not sure this is right:
* error from `Atoi` is ignored
* `Atoi` returns an `int` not an `int64` so if there's a problem it's already happened.
* `curState.rev` is already an `int64` so this cast is a no-op.

I think the right fix is to switch to `ParseInt( ..., 64)` and work entirely in `int64` for these revisions.